lib/foreman_tasks/dynflow/configuration.rb

```diff
@@ -14,19 +14,29 @@ def world_config
     end
 
     def backup_settings
+      return @backup_settings if @backup_settings
       backup_options = {
         :backup_deleted_plans => true,
         :backup_dir => default_backup_dir
       }
       settings = SETTINGS[:'foreman-tasks'] && SETTINGS[:'foreman-tasks'][:backup]
       backup_options.merge!(settings) if settings
-      backup_options
+      env_var = ENV['TASK_BACKUP']
+      unless env_var.nil?
+        backup_options[:backup_deleted_plans] = environment_override(env_var)
+      end
+      @backup_settings = backup_options
     end
 
     def default_backup_dir
       File.join(Rails.root, 'tmp', 'task-backup')
     end
 
+    def environment_override(env_var)
+      # Everything except 0, n, no, false is considered to be a truthy value
```

Comment:
Using `ENV['TASK_BACKUP']` directly here would be more readable, as it would encapsulate all we need to know about the influence of the env variable.